dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEnrollmentAnalyticsDimensionsService.java

```diff
@@ -77,24 +81,30 @@ public List<PrefixedDimension> getQueryDimensionsByProgramId(String programId) {
                                 .collect(Collectors.toSet())),
                         getProgramStageDataElements(QUERY, program),
                         filterByValueType(
-                            QUERY,
-                            ofItemsWithProgram(
-                                program, getTeasIfRegistrationAndNotConfidential(program))))))
+                            QUERY, ofItemsWithProgram(program, getTeasIfRegistration(program))))))
         .orElse(List.of());
   }
 
   private Collection<PrefixedDimension> getProgramStageDataElements(
       DimensionsServiceCommon.OperationType operationType, Program program) {
     return program.getProgramStages().stream()
         .map(ProgramStage::getProgramStageDataElements)
-        .map(
-            programStageDataElements ->
-                filterByValueType(
-                    operationType, ofProgramStageDataElements(programStageDataElements)))
+        .map(psdes -> excludeIfSKipAnalytics(operationType, psdes))
+        .map(psdes -> filterByValueType(operationType, ofProgramStageDataElements(psdes)))
         .flatMap(Collection::stream)
         .collect(Collectors.toList());
   }
 
+  private Collection<ProgramStageDataElement> excludeIfSKipAnalytics(
```

Comment:
Use specific collection class in return types, e.g. List or Set. It is good to use the generic Collection type in method parameter when appropriate.

Reply:
done